pkg/dockerregistry/server/auth.go

```diff
@@ -343,12 +344,36 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				}
 				verifiedPrune = true
 			default:
+				if repoAccessVerified {
```

Comment:
I agree, it's better to check twice then optimize out a check that haven't been done.